frontend/src/components/node/Details.tsx

```diff
@@ -119,6 +132,15 @@ export default function NodeDetails() {
     }, 1000);
   }
 
+  function handleCordonDialog() {
+    setCordonDialogOpen(!cordonDialogOpen);
+    console.log('cordonDialogOpen is now', cordonDialogOpen);
+  }
+
+  function handleDrainDialog() {
+    setDrainDialogOpen(!drainDialogOpen);
```

Comment:
The way to handle reversing the state properly is to use the function variant: `setDrainDialogOpen(openState => !openState)`